src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -480,12 +481,15 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
         // eslint-disable-next-line
     }, [results, filter]);
 
-    const viewRoom = (roomId: string, persist = false, viaKeyboard = false) => {
+    const viewRoom = (room: {room_id: string}|{room_alias: string}, persist = false, viaKeyboard = false) => {
         if (persist) {
+            if ('room_alias' in room) {
+                throw Error('Cannot persist room if room_alias is given');
+            }
```

Comment:
The caller on L612 always passes persist=true so will always cause this throw to happen

Reply:
We now allow persisting as we always pass in the roomId